src/installer/tests/nosdk/TempRoot.cs

```diff
@@ -0,0 +1,83 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+// Copied from dotnet/roslyn
+
+#nullable disable
+
+using System;
+using System.Collections.Concurrent;
+using System.IO;
+using System.Runtime.CompilerServices;
+
+namespace Microsoft.CodeAnalysis.Test.Utilities
+{
+    public sealed class TempRoot : IDisposable
```

Comment:
Could you please look if it made sense to share/unify with TestArtifact?
TestArtifact has support for optionally leaving files behind (to diagnose problems). It also got lot of testing in terms of robustness of delete and other IO operations.

Reply:
Sure. I copied this out of Roslyn since I knew where that one was, so this one should also be pretty solid. I'll try to unify.